### ISSUE TITLE
how to debug bluetooth problem for mybmw connect to car's app function

### ISSUE THREAD

**Issue by galenzhao** [BimmerGestalt/AAIdrive#796]
Hi,

Today, I upgraded my phone to Android 14 QPR3,
build from LineageOS 21.

After the upgrade, the MyBMW app can no longer connect to the car's app function.
The app checkbox is not shown in the car. (All permissions are set correctly for the MyBMW app, and it worked well before the upgrade.)

Before the upgrade, everything worked well.
So, it may be a problem with the Bluetooth in the new Android system.

How can I debug the problem? Is there any checkpoint for MyBMW to make the app function connect to the car?
Thanks.

**Comment by Andrea535LT** [comment]
Following. Same issue with my S24 Ultra

**Comment by hufman** [comment]
Unfortunately, the MyBMW>Car connection is relatively opaque, and I'm not sure what the car looks for before showing the Apps checkbox. The only suggestion I've heard work for other people is to enable all the permissions and background execution options (especially for Samsung), which it sounds like you've done. I've heard other people work well on Android 14 after setting these settings, and my own Sony phone hasn't had any issue with Android 14.

If you'd like to debug further, I have [notes on how to patch the Connected/MyBMW apps](https://bimmergestalt.github.io/BMWConnectedAnalysis/analysis-approach/) to enable debug logging, which can be combined with decompilation to try to trace through what it's doing. I haven't experienced the issue of the Apps checkbox not showing, so I can't help in this regard.

**Comment by galenzhao** [comment]
> Unfortunately, the MyBMW>Car connection is relatively opaque, and I'm not sure what the car looks for before showing the Apps checkbox. The only suggestion I've heard works for other people is to enable all the permissions and background execution options (especially for Samsung), which it sounds like you've done. I've heard other people work well on Android 14 after setting these settings, and my own Sony phone hasn't had any issues with Android 14.

I know the permission setting problem on Android 14, when I first upgraded to Android 14 beta I found it needed to give nearby device permission to the mybmw app which may be needed by connecting a Bluetooth device from the mybmw app.
Then all things going well.

> 
> If you'd like to debug further, I have [notes on how to patch the Connected/MyBMW apps](https://bimmergestalt.github.io/BMWConnectedAnalysis/analysis-approach/) to enable debug logging, which can be combined with decompilation to try to trace through what it's doing. I haven't experienced the issue of the Apps checkbox not showing, so I can't help in this regard.

Thanks for the info.
Cuz I'm using a self-build AOSP which may be the QPR3 version missing some Bluetooth functions.
I'm going to instigate the communication between mybmw and car and will come back if any progress.

**Comment by galenzhao** [comment]
maybe useful info from Logcat:

1719818404.550  1002  2610  3070 W ActiveDeviceManager: A2DP connection timeout. Activate HFP for XX:XX:XX:XX:D9:4F

**Comment by galenzhao** [comment]
is there any way to bypass the login view of BMW Connected_6.4.1.6349_Apkpure.apk
I followed the info and opened the log, but the app needed a login to open.

**Comment by hufman** [comment]
BMW Connected will connect to the car even without logging in, and even add its old Audioplayer and Calendar apps.

**Comment by galenzhao** [comment]
> BMW Connected will connect to the car even without logging in, and even add its old Audioplayer and Calendar apps.

yes,

and I just confirm that the old 'BMW Connected_6.4.1.6349_Apkpure.apk' can works without problem.

so it maybe new MyBMW and Android 14 QPR3 has something wrong.

**Comment by hufman** [comment]
Oh sorry, I missed this email:
The only negative effect I've seen is that older headunits get confused about both BMW Connected and MyBMW connecting at the same time and crash and restart. I've mainly seen this with the USB connection and IDrive4, but perhaps they are not correlated together.

**Comment by obhammer** [comment]
> I know the permission setting problem on Android 14, when I first upgraded to Android 14 beta I found it needed to give **_nearby device permission_** to the mybmw app which may be needed by connecting a Bluetooth device from the mybmw app. Then all things going well.
> 
NB! The newest release of My BMW (4.9.2 release date Sept 11 2024) resets all the given permissions for the app - and "Nearby devices" permission brakes the AAIdrive connection !!! Looks like all affected users need to manually give permission for Nearby devices in order to get the "App" show up in the iDrive bluetooth device properties, for making the AAIdrive work again

**Comment by hufman** [comment]
I'm not sure if an app upgrade would reset permissions, as I understand the Android upgrade process.
I do know Android has a feature to clear permissions of apps that aren't used in a long time, for "security". Perhaps this is what people have been experiencing?

**Comment by obhammer** [comment]
> I'm not sure if an app upgrade would reset permissions, as I understand the Android upgrade process. I do know Android has a feature to clear permissions of apps that aren't used in a long time, for "security". Perhaps this is what people have been experiencing?

Yes, it might have been a Android security updatethat did it as well - anyhow, I discovered this recently, so please check this if you encounter problems.  (it might be that timeout you're referring to as well - although I doubt that, since I use My BMW every day)
